### PR TITLE
Refactor

### LINKED ISSUE #56: rework language support files
pass node instead of text

### PR DESCRIPTION
**Pull request by jonhue**
Resolves #56

### CODE REVIEW

**Comment by jonhue** [comment]
Ready for review 🎉

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -5,19 +5,19 @@ on:
...
+  #   steps:
+  #     - uses: actions/checkout@master
+  #     - uses: actions/setup-node@v1
+  #       with:
+  #         node-version: 13
+  #         registry-url: https://npm.pkg.github.com/
+  #         scope: '@jonhue'
+  #     - run: npm publish
+  #       env:
+  #         NODE_AUTH_TOKEN: ${{ secrets.GPR_AUTH_TOKEN }}
```
Comment (antonbaumann) [inline .github/workflows/release.yml:20]: do we need this later?

File: src/@types/tree-sitter-go/index.d.ts [hunk 2]
```diff
@@ -1 +1 @@
-declare module 'tree-sitter-go';
\ No newline at end of file
+declare module 'tree-sitter-go'
```
Comment (antonbaumann) [inline src/@types/tree-sitter-go/index.d.ts]: missing semicolon
or did we change linter preferences?

File: src/WhoWroteThat.ts [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+                                    weight(this.strategy, owners)
+                                )
+                            ]));
+                    });
+            }).catch(() => {
+                // file could not be read at HEAD~commitIndex
+                return [];
+            });
+    }
+}
```
Comment (antonbaumann) [inline src/WhoWroteThat.ts:98]: 👍 👍

File: src/WhoWroteThat.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+const transformResult = (result: CodeOwners): CodeOwners => ({
+    ...result,
+    owners: squish(
+        result.owners
+            .sort((a, b) => a.score < b.score ? 1 : -1)
+    ).filter(owner => owner.score > 0)
+});
```
Comment (antonbaumann) [inline src/WhoWroteThat.ts]: method name not very descriptive
maybe rename or add a short comment

File: src/WhoWroteThat.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+const scale = (
+    strategy: Strategy,
+    owners: Owner[],
+    weight: number
+): Owner[] => {
+    if (strategy === Strategy.WEIGHTED_LINES)
+        return owners.map(owner => ({
+            ...owner,
+            score: owner.score * (weight - weight / 4) / weight
```
Comment (antonbaumann) [inline src/WhoWroteThat.ts:116]: why especially 4?

File: src/languages/Go.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+import TreeSitterGo from 'tree-sitter-go';
+import { Declaration } from '../types';
+
+const extractTypeDeclarationName = (node: Parser.SyntaxNode): string => {
+    return node.text.replace('type ', '').split(' ')[0];
+};
+
+const extractFunctionDeclarationName = (node: Parser.SyntaxNode): string => {
+    return node.text.replace('func ', '').split(' ')[0];
+};
```
Comment (antonbaumann) [inline src/languages/Go.ts:11]: `func (h *handler) getExapmpleByID(id int)`
would return `(h`

but we can fix that in another PR

File: src/languages/Java.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import Parser from 'tree-sitter';
+import TreeSitterJava from 'tree-sitter-java';
+import { Declaration } from '../types';
+
+const extractClassDeclarationName = (node: Parser.SyntaxNode): string => {
+    const indexClassToken = node.text.indexOf('class');
+    const elements = node.text.substring(indexClassToken + 6).split(' ');
```
Comment (antonbaumann) [inline src/languages/Java.ts:8]: we should at least write a comment why we use `+ 6`

**Comment by antonbaumann** [comment]
well done ❤️🎉

File: .github/workflows/release.yml (hunk 1 above)
Comment (jonhue) [reply .github/workflows/release.yml:20]: At some point we may want to publish this package to the GitHub Package Registry. So I'd just keep it for now.

File: src/languages/Go.ts (hunk 6 above)
Comment (jonhue) [reply src/languages/Go.ts:11]: That should be fixed once we move to tree-sitter fields.

File: src/WhoWroteThat.ts (hunk 5 above)
Comment (jonhue) [reply src/WhoWroteThat.ts:116]: This is just how I defined this strategy 😄. I'm going to document it with #57

**Review by antonbaumann — APPROVED** [review]

**Pull request merged**